src/context.cr

```diff
@@ -0,0 +1,118 @@
+
+require "./window.cr"
+
+module SDL
+
+	class GLContext
+	
+		alias Attribute = LibSDL::GLattr
+		alias Profile = LibSDL::GLprofile
+		alias Flag = LibSDL::GLcontextFlag
+
+		@@lib_loaded = false
+
+
+		def self.load_library( path : String | Nil ) : Nil
+			return if @@lib_loaded
+
+			ret = 0
```

Comment:
You may just do the following; Crystal automatically transforms `nil` to a NULL pointer, and specifying types is optional.

```crystal
def self.load_library(path)
  return if @@initialized
  @@initialized = true

  if LibSDL.gl_load_library(path) == 0
    raise Error.new("SDL_GL_LoadLibrary")
  end
end
```